pkg/templates/privilegeescalation/template.go

```diff
@@ -0,0 +1,38 @@
+package privilegeescalation
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/privilegeescalation/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Privilege Escalation on Containers",
+		Key:         "privilege-escalation-container",
+		Description: "Flag containers of allowing privilege escalation",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+				securityContext := container.SecurityContext
```

Comment:
Per the documentation:
```
	// AllowPrivilegeEscalation controls whether a process can gain more
	// privileges than its parent process. This bool directly controls if
	// the no_new_privs flag will be set on the container process.
	// AllowPrivilegeEscalation is true always when the container is:
	// 1) run as Privileged
	// 2) has CAP_SYS_ADMIN
```
Therefore, I think if the container is Privileged, _or_ has CAP_SYS_ADMIN, we should still fail this check (with the appropriate message). What do you think?